core/trino-main/src/main/java/io/trino/sql/planner/planprinter/anonymize/AnonymizePlanPrinter.java

```diff
@@ -0,0 +1,398 @@
+
+import com.google.common.base.Joiner;
+import io.airlift.json.JsonCodec;
+import io.trino.Session;
+import io.trino.metadata.Catalog;
+import io.trino.metadata.Metadata;
+import io.trino.metadata.TableSchema;
+import io.trino.spi.connector.ColumnSchema;
+import io.trino.sql.planner.OrderingScheme;
+import io.trino.sql.planner.Partitioning;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.plan.AggregationNode;
+import io.trino.sql.planner.plan.ExchangeNode;
+import io.trino.sql.planner.plan.FilterNode;
+import io.trino.sql.planner.plan.JoinNode;
+import io.trino.sql.planner.plan.PlanNode;
+import io.trino.sql.planner.plan.PlanNodeId;
+import io.trino.sql.planner.plan.PlanVisitor;
+import io.trino.sql.planner.plan.ProjectNode;
+import io.trino.sql.planner.plan.RemoteSourceNode;
+import io.trino.sql.planner.plan.SemiJoinNode;
+import io.trino.sql.planner.plan.SortNode;
+import io.trino.sql.planner.plan.TableScanNode;
+import io.trino.sql.planner.plan.ValuesNode;
+import io.trino.sql.planner.plan.WindowNode;
+import io.trino.sql.planner.planprinter.PlanPrinter;
+import io.trino.sql.tree.BinaryLiteral;
+import io.trino.sql.tree.CharLiteral;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.ExpressionRewriter;
+import io.trino.sql.tree.ExpressionTreeRewriter;
+import io.trino.sql.tree.GenericLiteral;
+import io.trino.sql.tree.Literal;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.sql.tree.SymbolReference;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.trino.sql.planner.Partitioning.ArgumentBinding;
+import static io.trino.sql.planner.plan.AggregationNode.Aggregation;
+import static io.trino.sql.planner.plan.WindowNode.Function;
+import static io.trino.sql.planner.plan.WindowNode.Specification;
+import static io.trino.sql.planner.planprinter.PlanPrinter.unresolveFunctions;
+import static io.trino.sql.planner.planprinter.anonymize.AggregationNodeRepresentation.AggregationRepresentation;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class AnonymizePlanPrinter
+{
+    private enum ObjectType
+    {
+        CATALOG,
+        SCHEMA,
+        TABLE,
+        COLUMN,
+        SYMBOL,
+        LITERAL
+    }
+
+    private final Metadata metadata;
+    private final Session session;
+    private final JsonCodec<AnonymizedNodeRepresentation> representationJsonCodec = jsonCodec(AnonymizedNodeRepresentation.class);
+
+    private final Map<PlanNodeId, AnonymizedNodeRepresentation> representations = new HashMap<>();
```

Comment:
This map and `addRepresentation` is unnecessary since you're just serialising from the top most node recursively.